auth/src/main/java/org/aerogear/auth/impl/KeycloakUserPrincipalImpl.java

```diff
@@ -0,0 +1,76 @@
+package org.aerogear.auth.impl;
+
+
+import org.aerogear.auth.AbstractAuthenticator;
+import org.aerogear.auth.IKeycloakUserPrincipal;
+import org.aerogear.auth.IRole;
+import org.aerogear.auth.credentials.OIDCCredentials;
+
+import java.util.Map;
+
+public class KeycloakUserPrincipalImpl extends UserPrincipalImpl implements IKeycloakUserPrincipal {
```

Comment:
all those fields are not specific to Keycloak, they are all part of the OIDC protocol. Let's rename this to `OIDCUserPrinicipalImpl`